### PR TITLE
sanitycheck: add "ASSERTION FAIL" to list of error messages caught

### PR DESCRIPTION
**Pull request by marc-hb**
Assertions should not be just for developers, CI should catch them too.

This would have caught earlier issue #17555 (fixed by PR ~#17585~ #17599)

As usual, invidual tests can opt out with the "ignore_faults" tag.

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by andrewboie** [comment]
CI failures will go away with #17170 merged

**Comment by marc-hb** [comment]
> CI failures will go away with #17170 merged

I had at look at the failures and I'm pretty sure you meant PR #17585 . Will close/re-open once it is merged.

Unless I missed some, all the failures in https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/46087 are the expected `ASSERTION FAIL [plog_backend_count_get() > 0osix_fs_test]` which proves this PR is a "successfully tested test".

**Comment by marc-hb** [comment]
> CI failures will go away with #17170 merged

Superseded by PR #17599

**Comment by ioannisg** [comment]
Can we close this PR?

**Comment by marc-hb** [comment]
> Can we close this PR?

Because PR #17544 made it unnecessary? Trying to figure out whether it did...

**Comment by marc-hb** [comment]
I can get a timeout  now (HEAD= bd72ea1df9d0, including PR #17544) when I artificially trigger the ASSERT that started all this 
```diff
--- a/subsys/logging/log_core.c
+++ b/subsys/logging/log_core.c
@@ -867,7 +867,7 @@ void log_free(void *str)
 
 static void log_process_thread_func(void *dummy1, void *dummy2, void *dummy3)
 {
-	__ASSERT_NO_MSG(log_backend_count_get() > 0);
+	__ASSERT_NO_MSG(log_backend_count_get() > 42);
 
 	log_init();
 	thread_set(k_current_get());
```
```
time $ZEPHYR_BASE/scripts/sanitycheck -T $ZEPHYR_BASE/tests/kernel/common/ -p qemu_x86_64 -vv
1/2 qemu_x86_64               tests/kernel/common/kernel.common.misra            FAILED: timeout (qemu 60.000s)
	see: sanity-out/qemu_x86_64/tests/kernel/common/kernel.common.misra/handler.log
```
According to commit f2c849b2c918 sanitycheck now looks for `ZEPHYR FATAL ERROR` and it does get printed promptly yet this particular test times out:
 ```
starting test - test_timeout_order
ASSERTION FAIL [log_backend_count_get() > 32] @ ZEPHYR_BASE/subsys/logging/log_core.c:870
FATAL: *** FATAL ERROR vector 5 code 1016
FATAL: ***  RIP 8:0x105192 RSP 16:0x1135c8 RFLAGS 0x202
FATAL: ***  RAX 0x10d6f3 RCX 0xa RDX 0x3f8 RSI 0x366 RDI 0x10e7a0
FATAL: ***  R8 0xffffffff R9 0x0 R10 0x0 R11 0x0
FATAL: >>> ZEPHYR FATAL ERROR 4: Kernel panic
FATAL: Current thread: 0x0010fe20 (logging)
FATAL: Halting system
```
Other tests fail immediately.

@andrewboie can you shed some light?

**Comment by marc-hb** [comment]
Re-opening not to lose track of the question above. Remove yourself from the PR if not interested.

**Comment by nashif** [comment]
@marc-hb  are you going to move this forward?

**Comment by marc-hb** [comment]
This is now a question/bug about @andrewboie's completely different fix but I will first try to reproduce with the latest master. Thanks for the reminder.

PS: don't waste your precious maintainer time looking at github Drafts.

**Comment by marc-hb** [comment]
It took me a while because the bug moved a bit to evade me but the timeout is still there and apparently the same. @andrewboie do you want a new github issue or is this something you can fix faster than I file it?

 `log_process_thread_func()` doesn't get call by default any more, so now the reproduction requires something like this:
```diff
--- a/subsys/logging/log_core.c
+++ b/subsys/logging/log_core.c
@@ -1159,6 +1159,7 @@ struct k_thread logging_thread;
 static int enable_logger(struct device *arg)
 {
        ARG_UNUSED(arg);
+       __ASSERT_NO_MSG(0);
 
        if (IS_ENABLED(CONFIG_LOG_PROCESS_THREAD)) {
                /* start logging thread */
```
Then the error handling is quite inconsistent:
```
time $ZEPHYR_BASE/scripts/sanitycheck -T $ZEPHYR_BASE/tests/kernel/common/ -v -p native_posix -p native_posix_64 -p mps2_an385 -p mps2_an521 -p qemu_x86 -p qemu_x86_64 -p qemu_x86_coverage -p qemu_cortex_m3 -p qemu_x86_nommu -p qemu_nios2 -p qemu_x86_long -p qemu_xtensa

22 tests selected, 382 tests discarded due to filters
 1/22 native_posix              tests/kernel/common/kernel.common.misra            FAILED: error (native 0.006s)
	see: sanity-out/native_posix/tests/kernel/common/kernel.common.misra/handler.log
 2/22 native_posix_64           tests/kernel/common/kernel.common.misra            FAILED: error (native 0.006s)
	see: sanity-out/native_posix_64/tests/kernel/common/kernel.common.misra/handler.log
 3/22 native_posix_64           tests/kernel/common/kernel.common                  FAILED: error (native 0.004s)
	see: sanity-out/native_posix_64/tests/kernel/common/kernel.common/handler.log
 4/22 native_posix              tests/kernel/common/kernel.common                  FAILED: error (native 0.008s)
	see: sanity-out/native_posix/tests/kernel/common/kernel.common/handler.log
 5/22 qemu_x86                  tests/kernel/common/kernel.common                  FAILED: handler_crash (qemu)
	see: sanity-out/qemu_x86/tests/kernel/common/kernel.common/handler.log
 6/22 qemu_x86                  tests/kernel/common/kernel.common.misra            FAILED: handler_crash (qemu)
	see: sanity-out/qemu_x86/tests/kernel/common/kernel.common.misra/handler.log
 7/22 qemu_cortex_m3            tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.059s)
	see: sanity-out/qemu_cortex_m3/tests/kernel/common/kernel.common/handler.log
 8/22 qemu_nios2                tests/kernel/common/kernel.common.misra            FAILED: timeout (qemu 60.059s)
	see: sanity-out/qemu_nios2/tests/kernel/common/kernel.common.misra/handler.log
 9/22 qemu_nios2                tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.059s)
	see: sanity-out/qemu_nios2/tests/kernel/common/kernel.common/handler.log
10/22 mps2_an385                tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.059s)
	see: sanity-out/mps2_an385/tests/kernel/common/kernel.common/handler.log
11/22 qemu_cortex_m3            tests/kernel/common/kernel.common.misra            FAILED: timeout (qemu 60.059s)
	see: sanity-out/qemu_cortex_m3/tests/kernel/common/kernel.common.misra/handler.log
12/22 mps2_an521                tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.059s)
	see: sanity-out/mps2_an521/tests/kernel/common/kernel.common/handler.log
13/22 qemu_x86_coverage         tests/kernel/common/kernel.common                  FAILED: handler_crash (qemu)
	see: sanity-out/qemu_x86_coverage/tests/kernel/common/kernel.common/handler.log
14/22 qemu_x86_64               tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.058s)
	see: sanity-out/qemu_x86_64/tests/kernel/common/kernel.common/handler.log
15/22 qemu_x86_nommu            tests/kernel/common/kernel.common                  FAILED: handler_crash (qemu)
	see: sanity-out/qemu_x86_nommu/tests/kernel/common/kernel.common/handler.log
16/22 qemu_x86_nommu            tests/kernel/common/kernel.common.misra            FAILED: handler_crash (qemu)
	see: sanity-out/qemu_x86_nommu/tests/kernel/common/kernel.common.misra/handler.log
17/22 qemu_x86_coverage         tests/kernel/common/kernel.common.misra            FAILED: handler_crash (qemu)
	see: sanity-out/qemu_x86_coverage/tests/kernel/common/kernel.common.misra/handler.log
18/22 qemu_x86_64               tests/kernel/common/kernel.common.misra            FAILED: timeout (qemu 60.059s)
	see: sanity-out/qemu_x86_64/tests/kernel/common/kernel.common.misra/handler.log
19/22 qemu_x86_long             tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.058s)
	see: sanity-out/qemu_x86_long/tests/kernel/common/kernel.common/handler.log
20/22 qemu_x86_long             tests/kernel/common/kernel.common.misra            FAILED: timeout (qemu 60.059s)
	see: sanity-out/qemu_x86_long/tests/kernel/common/kernel.common.misra/handler.log
21/22 qemu_xtensa               tests/kernel/common/kernel.common                  FAILED: timeout (qemu 60.060s)
	see: sanity-out/qemu_xtensa/tests/kernel/common/kernel.common/handler.log
22/22 qemu_xtensa               tests/kernel/common/kernel.common.misra            FAILED: timeout (qemu 60.060s)
	see: sanity-out/qemu_xtensa/tests/kernel/common/kernel.common.misra/handler.log
0 of 22 tests passed with 0 warnings in 156 seconds
```

**Comment by marc-hb** [comment]
Filed issue #19492

**Pull request closed without merging**
